src/pybind/mgr/dashboard_v2/models/cluster.py

```diff
@@ -80,9 +77,8 @@ def save(self, force_insert=False, force_update=False, using=None,
                 for flag in set(value) - set(original.osd_flags):
                     api.osd_set(flag)
             else:
-                pass
-                #  logger.warning('Tried to set "{}" to "{}" on rbd "{}", which is not '
-                #               'supported'.format(key, value, self.config_file_path))
+                logger.warning('Tried to set "{}" to "{}" on rbd "{}", which is not '
+                               'supported'.format(key, value, self.config_file_path))
```

Comment:
Same here, remove the `format` call.

Reply:
why?

Reply:
In my example that I showed above (in other comment), the string composition/concatenation is only preformed inside the `debug` method, which will only run if the `DEBUG` level is activated, opposed to your code where we will always incur in an extra overhead of constructing the log message string even if the `DEBUG` level is not active.

That is why that pylint by default warns when someone uses the `format()` to build a log message string.

Reply:
Thy pylint warning doesn't make much sense.

`.debug()` hides errors. For example, this will print an exception only if you enable debugging:

```python
logging.debug('%', 'ricardo')
```
While this fails regardless of the log level:
```python
logging.debug('{'.format('ricardo'))
```
And I'm very much in favor of trading correctness and testablility versus a minor performance improvement. 

In your concrete example here, the string will be of course generated regardless.

And finally, I doubt that we will ever have a performance problem that is caused by a log message in a tight loop. Performance of the Python code itself is in general not an issue for us. Instead correctness, the number of network requests and context switches is much more important.

Reply:
Almost every logging library for any programming language that I know of allows the construction of the message string to be made inside the logging function.
That's the pattern/convention that most programmer are used to use when adding logging messages to code, so if you don't mind, please do the same to match the way logging functions are used in other parts of this project.

Reply:
I don't think that's a valid point. I don't see a reason for changing existing code for nothing.

Reply:
If you don't consider "uniform use of logging functions across the project" and "performance issues" as valid points, then we shouldn't continue this discussion.
Moreover, you are not changing existing code, that code is not part of the project yet.

Reply:
Looks like there is no strict enforcement of logging:

* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L64
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L79
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L97
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L621
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L218
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L121-L122
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L1011
* https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/dashboard/module.py#L1235-L1236

I have to agree with all `%`  and `.format` usages in the exsisting dashboard: That code is a tiny bit more tested this way. 
And that code is an import of an existing stand alone openATTIC project. Everytime we modify something when importing stuff, we introduce new bugs. Expecially as (as least I am) in a hurry to get things done.

Reply:
Alright, it will not be because of me that the development does not progress.
Don't mind my issues when merging this PR.
I'll open a PR later in the future to address these concerns.

Reply:
> I'll open a PR later in the future to address these concerns.
That would be cool!